src/libcore/unicode/unicode.py

```diff
@@ -1,93 +1,286 @@
 #!/usr/bin/env python
 
-# This script uses the following Unicode tables:
-# - DerivedCoreProperties.txt
-# - DerivedNormalizationProps.txt
-# - EastAsianWidth.txt
-# - auxiliary/GraphemeBreakProperty.txt
-# - PropList.txt
-# - ReadMe.txt
-# - Scripts.txt
-# - UnicodeData.txt
-#
-# Since this should not require frequent updates, we just store this
-# out-of-line and check the tables.rs file into git.
+"""
+Regenerate Unicode tables (tables.rs).
+"""
 
-import fileinput, re, os, sys, operator, math, datetime
+# This script uses the Unicode tables as defined
+# in the UnicodeFiles class.
 
-# The directory in which this file resides.
-fdir = os.path.dirname(os.path.realpath(__file__)) + "/"
+# Since this should not require frequent updates, we just store this
+# out-of-line and check the tables.rs file into git.
 
-preamble = '''
+# Note that the "curl" program is required for operation.
+# This script is compatible with Python 2.7 and 3.x.
```

Comment:
very minor nit: is there a reason these comments aren't part of the doc comment?

Reply:
The doc comment serves dual purpose: it describes the script's purpose and serves as the `argparse` program description (which one can see when running the script with `--help` flag). 

Compatibility notice wouldn't be very helpful once the program runs, and it would probably not even show up if incompatible Python version was used. That being said, the changes I made were meant to support current and future Python 3 versions alike, and the script has no external dependencies outside of the standard library.

Similarly, for curl, the script would just crash if curl wasn't installed. In the example below, I have modified `curl` to read `córl` and the following traceback was printed:

```
Fetching: ftp://ftp.unicode.org/Public/UNIDATA/ReadMe.txt
Traceback (most recent call last):
  File "...rust/src/libcore/unicode/unicode.py", line 868, in <module>
    main()
  File "...rust/src/libcore/unicode/unicode.py", line 812, in main
    unicode_version = fetch_files(args.version)
  File "...rust/src/libcore/unicode/unicode.py", line 147, in fetch_files
    readme_content = subprocess.check_output(("córl", readme_url))
  File "/usr/lib/python3.7/subprocess.py", line 395, in check_output
    **kwargs).stdout
  File "/usr/lib/python3.7/subprocess.py", line 472, in run
    with Popen(*popenargs, **kwargs) as process:
  File "/usr/lib/python3.7/subprocess.py", line 775, in __init__
    restore_signals, start_new_session)
  File "/usr/lib/python3.7/subprocess.py", line 1522, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
FileNotFoundError: [Errno 2] No such file or directory: 'córl': 'córl'
```

Whether curl should be replaced by Python's [ftplib](https://docs.python.org/3.7/library/ftplib.html) for maximum compatibility and no external dependencies is debatable -- `curl` is listed in Rust build dependencies.

So, given the above, I figured that a mere comment with a compatibility note would suffice, and its contents did not feel like they belonged to the docstring.

Reply:
thanks for your explanation!